framework/src/state_machine/state_machine.ts

```diff
@@ -177,15 +180,15 @@ export class StateMachine {
 				[ctx.transaction.id],
 			);
 		} catch (error) {
-			ctx.eventQueue.restoreSnapshot();
-			ctx.stateStore.restoreSnapshot();
+			ctx.eventQueue.restoreSnapshot(commandEventQueueSnapshotID);
+			ctx.stateStore.restoreSnapshot(commandStateStoreSnapshotID);
 			ctx.eventQueue.add(
 				ctx.transaction.moduleID,
 				EVENT_STANDARD_TYPE_ID,
 				codec.encode(standardEventDataSchema, { success: false }),
 				[ctx.transaction.id],
 			);
-			status = TransactionExecutionResult.FAIL;
+			return TransactionExecutionResult.FAIL;
```

Comment:
here actually we cannot return `FAIL` because even if it fails, we want to execute `afterCommandExecute`. I think that's why `OK` status was defined in the beginning.

Reply:
Makes sense if we want proceed with execution not to return 🙂